### PR TITLE
Bug 1404874 - [Regression] Context menu is triggered when single tapp…

### PR DESCRIPTION
**Pull request by justindarc**
…ing on some images

### CODE REVIEW

File: Client/Assets/ContextMenu.js [hunk 1]
```diff
@@ -1,175 +1,127 @@
...
+const GESTURE_ALLOWABLE_MOVEMENT = 10;
+const GESTURE_MINIMUM_PRESS_DURATION = 500;
 
-var MAX_RADIUS = 9;
+const SUPPRESS_NEXT_CLICK_DURATION = 50;
 
+var target = null;
+var originalTargetTouchCalloutStyle = '';
+
+var didLongPress = false;
```
Comment (jhugman) [inline Client/Assets/ContextMenu.js]: Nit: Use `let`. If we're moving to `const` we should just drop `var` entirely.

File: Client/Assets/ContextMenu.js [hunk 2]
```diff
@@ -1,175 +1,127 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+///* This Source Code Form is subject to the terms of the Mozilla Public
+// * License, v. 2.0. If a copy of the MPL was not distributed with this
+// * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (jhugman) [inline Client/Assets/ContextMenu.js]: Nit: wait, what?

File: Client/Assets/ContextMenu.js [hunk 3]
```diff
@@ -1,175 +1,130 @@
...
+  // context menu from appearing.
   setTimeout(function() {
-    if (element) {
-      element.removeEventListener("click", handleClick, true);
+    if (evt.defaultPrevented) {
+      cancel();
+      return;
     }
+
+    longPressTimeout = setTimeout(function() {
```
Comment (jhugman) [inline Client/Assets/ContextMenu.js]: Please could you put this function outside of this one?

File: Client/Assets/ContextMenu.js [hunk 4]
```diff
@@ -1,175 +1,130 @@
...
-  cancel();
+  touchstartScreenX = evt.touches[0].screenX;
+  touchstartScreenY = evt.touches[0].screenY;
 
-  window.removeEventListener("touchend", handleTouchEnd);
-  window.removeEventListener("touchmove", handleTouchMove);
+  didLongPress = false;
 
+  // Wait till the next tick before continuing to check for the gesture
+  // to give page scripts a change to `preventDefault()` to prevent the
```
Comment (boek) [inline Client/Assets/ContextMenu.js]: Did you mean change here?

File: Client/Assets/ContextMenu.js (hunk 1 above)
Comment (justindarc) [reply Client/Assets/ContextMenu.js]: I think we intend on uplifting this to `v9.x`, so we can't use `let` there (support began with iOS 10) :-/

https://caniuse.com/#feat=let

**Pull request merged**